### PR TITLE
DDS delete_topic [8180]

### PR DESCRIPTION
**Pull request by MiguelCompany**
This PR includes:
* Reference counting methods on TopicImpl
* Implementation and unit test for delete_topic API

### CODE REVIEW

File: src/cpp/fastdds/topic/TopicImpl.cpp [hunk 1]
```diff
@@ -41,6 +41,8 @@ TopicImpl::TopicImpl(
     , qos_(&qos == &TOPIC_QOS_DEFAULT ? participant_->get_default_topic_qos() : qos)
     , listener_(listen)
     , user_topic_(nullptr)
+    , handle_()
```
Comment (laura-eprosima) [inline src/cpp/fastdds/topic/TopicImpl.cpp:44]: This is not necessary. Topic inherits from DomainEntity which already has instance_handle_

Comment (MiguelCompany) [reply src/cpp/fastdds/topic/TopicImpl.cpp:44]: 🤔 It was there before this PR. I will remove it.

Comment (MiguelCompany) [reply src/cpp/fastdds/topic/TopicImpl.cpp]: Done in cf3e98a

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1907/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1645/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1900/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1908/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1646/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1901/)

**Pull request merged**